gax/src/streamingCalls/streaming.ts

```diff
@@ -522,4 +363,232 @@ export class StreamProxy extends duplexify implements GRPCCallResult {
+                'Cannot set both totalTimeoutMillis and maxRetries ' +
+                  'in backoffSettings.'
+              );
+              newError.code = Status.INVALID_ARGUMENT;
+              // clean up the request stream and retryStreams, silently destroy it on the request stream
+              // but do raise it on destructin of the retryStream so the consumer can see it
+              requestStream.destroy();
+              retryStream.destroy(newError);
+
+              return retryStream;
+            } else {
+              // check for exceeding timeout or max retries
+
+              try {
+                this.throwIfMaxRetriesOrTotalTimeoutExceeded(
+                  deadline,
+                  maxRetries!,
+                  timeout!,
+                  error,
+                  totalTimeout,
+                  retries
+                );
+              } catch (error: unknown) {
+                const e = GoogleError.parseGRPCStatusDetails(
+                  error as GoogleError
+                );
+                // clean up the request stream and retryStreams, silently destroy it on the request stream
+                // but do raise it on destruction of the retryStream so the consumer can see it
+                requestStream.destroy();
+                retryStream.destroy(e);
+
+                return retryStream;
+              }
+
+              const delayMult = retry.backoffSettings.retryDelayMultiplier;
+              const maxDelay = retry.backoffSettings.maxRetryDelayMillis;
+              const timeoutMult = retry.backoffSettings.rpcTimeoutMultiplier;
+              const maxTimeout = retry.backoffSettings.maxRpcTimeoutMillis;
+              let delay = retry.backoffSettings.initialRetryDelayMillis;
+              // calculate new deadlines
+              const toSleep = Math.random() * delay;
+              const calculateTimeoutAndResumptionFunction = () => {
+                setTimeout(() => {
+                  // only do timeout calculations if not using maxRetries
+                  if (timeout) {
+                    now = new Date();
+                    delay = Math.min(delay * delayMult, maxDelay);
+                    const timeoutCal =
+                      timeout && timeoutMult ? timeout * timeoutMult : 0;
+                    const rpcTimeout = maxTimeout ? maxTimeout : 0;
+                    const newDeadline = deadline ? deadline - now.getTime() : 0;
+                    timeout = Math.min(timeoutCal, rpcTimeout, newDeadline);
+                  }
+
+                  retries++;
+                  let retryArgument = this.argument! as unknown as RequestType;
+                  // if resumption logic is passed, use it to determined the
+                  // new argument for the new request made to the server
+                  // otherwise, the original argument will be passed
+                  if (typeof retry.getResumptionRequestFn! === 'function') {
```

Comment:
you shouldn't need to confirm the existence of retry.getResumptionRequestFn, since you're typechecking (and thus checking it's not null or undefined).